Many Changes with the timeseries dao. Now it is using odm2api.

lsetiawan opened the pull request:
Many Changes with the timeseries dao. Now it is using odm2api to do the queries and such. Also, implementation of DB configuration have started, now runserver.py reads from `.cfg` for DATABASE config. 

**NOTE: There are still some issues that needs to be addressed, such as error below when `WOFpy` is running**

```bash
Error running Query: (sqlite3.ProgrammingError) SQLite objects created in a thread can only be used in that same thread.The object was created in thread id 139731677263616 and this is thread id 139731685656320...
```
*I'll open up issue in a bit.. this pull request is mostly for me to see if `odm2_dao_sqlite_test` passes, due to the huge changes that is about to happen on the `ODM2PythonAPI` side, I would not merge yet.* Thanks.

ocefpaf commented on test/test_odm2_dao_sqlite.py:
```diff
@@ -5,7 +5,8 @@
 from wof.examples.flask.odm2.timeseries.odm2_timeseries_dao import Odm2Dao as OdmDao  # noqa
 
-ODM2_DATABASE_URI = 'sqlite:///' + "./test/odm2/ODM2.sqlite"
+# ODM2_DATABASE_URI = 'sqlite:///' + "./test/odm2/ODM2.sqlite"
```
Once you are done consider removing the commented out code. There are many lines in WOFpy like that and they can be very confusing and noisy for a newcomer (or even an experienced WOFpy dev at a future time).

lsetiawan replied:
@ocefpaf Ah, yes. Will do. Thanks. Looks like the changes in the dao passes now! 😄

emiliom commented:
> Looks like the changes in the dao passes now!

Are you saying the odm2 timeseries dao test passes with a SQLite ODM2 database?! If so -- woo-hoo!

lsetiawan commented:
> Are you saying the odm2 timeseries dao test passes with a SQLite ODM2 database?! If so -- woo-hoo!

It appears to be that way based on Travic CI reports. Though I hope it won't break too much with the massive changes on the odm2api side.

emiliom commented:
> I'll open up issue in a bit.. 

Were you still planning to open a new issue? I see that you made a related comment in issue #75 the same day.

> this pull request is mostly for me to see if odm2_dao_sqlite_test passes, due to the huge changes that is about to happen on the ODM2PythonAPI side, I would not merge yet. 

Just so I'm clear: you are trying to get `test_odm2_dao_sqlite.py` to work with the current `odm2api`, so we're better prepared once the new, large `odm2api` PR is merged (and the odm2api conda package is updated)?

Have you checked that your changes don't break the odm2 timeseries DAO with the LBR MySQL sample DB? (I know you mentioned tests in other DB's in issue #75).

Thanks!

lsetiawan commented:
> Were you still planning to open a new issue? I see that you made a related comment in issue #75 the same day.

I was going to open up a new issue, but I realized that there was already another issue that was related to this stuff. That's why I just added it there. 

> Just so I'm clear: you are trying to get test_odm2_dao_sqlite.py to work with the current odm2api, so we're better prepared once the new, large odm2api PR is merged (and the odm2api conda package is updated)?

I've modified the `odm2_timeseries_dao.py` to use `odm2api` rather than straight up sqlalchemy queries. This is why I'm worried if any of the changes on the `odm2api` side would effect the DAO again.

> Have you checked that your changes don't break the odm2 timeseries DAO with the LBR MySQL sample DB? 

I am now using `odm2api` so my thought is that since that works for all 3, then nothing will break. Though I still need to do further test before saying yes for sure.

emiliom commented:
> I've modified the odm2_timeseries_dao.py to use odm2api rather than straight up sqlalchemy queries. 

Ok. As you know, `odm2_timeseries_dao.py` already uses some parts of `odm2api`:
```python
from odm2api.ODM2 import models as odm2_models
```
But yes, it doesn't use `odm2api` query machinery.

> This is why I'm worried if any of the changes on the odm2api side would effect the DAO again.

Understood. But we either merge or don't merge your PR (or tweak it)! So again I'm not sure if you're saying you ultimately will try to merge these changes before the new `odm2api` are in place ... or not. Is your plan to withdraw this PR after you've run some tests?

lsetiawan commented:
> But we either merge or don't merge your PR (or tweak it)! So again I'm not sure if you're saying you ultimately will try to merge these changes before the new odm2api are in place ... or not. Is your plan to withdraw this PR after you've run some tests?

My thoughts are not to merge this PR before the new odm2api are in place, and see if anything breaks. If somethings break then I will withdraw this PR, and tweak it, also so some testing on MySQL, then do the PR again. I really need to find a clean PostgreSQL database to test on.. Thanks.

emiliom commented:
Ok.

ocefpaf commented:
> My thoughts are not to merge this PR before the new odm2api are in place, and see if anything breaks.

We should improve the test with any breakage you may find. The test coverage is very poor at the moment. (A green Travis-CI does not mean much.)

> If somethings break then I will withdraw this PR, and tweak it, also so some testing on MySQL, then do the PR again. I really need to find a clean PostgreSQL database to test on. Thanks.

Just my two cents. If things are OK with the current version of `odm2api` we should pin to that version and worry later about the massive changes that is underway.
